internal/processing/status.go

```diff
@@ -0,0 +1,96 @@
+package processing
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	gatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+	"github.com/kyma-incubator/api-gateway/internal/validation"
+)
+
+type ReconciliationStatus struct {
+	ApiRuleStatus               *gatewayv1beta1.APIRuleResourceStatus
+	VirtualServiceStatus        *gatewayv1beta1.APIRuleResourceStatus
+	AccessRuleStatus            *gatewayv1beta1.APIRuleResourceStatus
+	RequestAuthenticationStatus *gatewayv1beta1.APIRuleResourceStatus
+	AuthorizationPolicyStatus   *gatewayv1beta1.APIRuleResourceStatus
+}
+
+func getStatus(apiStatus *gatewayv1beta1.APIRuleResourceStatus, statusCode gatewayv1beta1.StatusCode) ReconciliationStatus {
```

Comment:
This will be handled properly in this task:

- https://github.com/kyma-project/api-gateway/issues/84